WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SignupGoogleFragment.java

```diff
@@ -0,0 +1,194 @@
+                        }
+                    } else {
+                        AnalyticsTracker.track(AnalyticsTracker.Stat.LOGIN_SOCIAL_BUTTON_FAILURE);
+                        switch (signInResult.getStatus().getStatusCode()) {
+                            // Internal error.
+                            case GoogleSignInStatusCodes.INTERNAL_ERROR:
+                                AppLog.e(T.NUX, "Google Signup Failed: internal error.");
+                                showErrorDialog(getString(R.string.login_error_generic));
+                                break;
+                            // Attempted to connect with an invalid account name specified.
+                            case GoogleSignInStatusCodes.INVALID_ACCOUNT:
+                                AppLog.e(T.NUX, "Google Signup Failed: invalid account name.");
+                                showErrorDialog(getString(R.string.login_error_generic)
+                                        + getString(R.string.login_error_suffix));
+                                break;
+                            // Network error.
+                            case GoogleSignInStatusCodes.NETWORK_ERROR:
+                                AppLog.e(T.NUX, "Google Signup Failed: network error.");
+                                showErrorDialog(getString(R.string.error_generic_network));
+                                break;
+                            // Cancelled by the user.
+                            case GoogleSignInStatusCodes.SIGN_IN_CANCELLED:
+                                AppLog.e(T.NUX, "Google Signup Failed: cancelled by user.");
+                                break;
+                            // Attempt didn't succeed with the current account.
+                            case GoogleSignInStatusCodes.SIGN_IN_FAILED:
+                                AppLog.e(T.NUX, "Google Signup Failed: current account failed.");
+                                showErrorDialog(getString(R.string.login_error_generic));
+                                break;
+                            // Attempted to connect, but the user is not signed in.
+                            case GoogleSignInStatusCodes.SIGN_IN_REQUIRED:
+                                AppLog.e(T.NUX, "Google Signup Failed: user is not signed in.");
+                                showErrorDialog(getString(R.string.login_error_generic));
+                                break;
+                            // Unknown error.
+                            default:
+                                AppLog.e(T.NUX, "Google Signup Failed: unknown error.");
+                                showErrorDialog(getString(R.string.login_error_generic));
+                                break;
+                        }
+                    }
+                } else if (result == RESULT_CANCELED) {
+                    AnalyticsTracker.track(AnalyticsTracker.Stat.LOGIN_SOCIAL_BUTTON_FAILURE);
```

Comment:
We're tracking `LOGIN_SOCIAL_BUTTON_FAILURE` with no extra data both here and below, so we won't be able to tell if the user cancelled or something else went wrong. Should we be doing that?

Reply:
Unfortunately, `RESULT_CANCELED` can be multiple things (e.g. canceled by the user intentionally or canceled by the Google API due to some unknown reason).  We don't get anything other than the result value so we can't really handle it any differently than an unknown error like in the `else` case below.